lib/Mojo/mysql/Native/Database.pm

```diff
@@ -0,0 +1,227 @@
+package Mojo::mysql::Native::Database;
+use Mojo::Base 'Mojo::mysql::Database';
+
+use Mojo::mysql::Connection;
+use Mojo::mysql::Native::Results;
+use Mojo::mysql::Native::Transaction;
+use Mojo::mysql::Util qw(expand_sql parse_url);
+use Scalar::Util 'weaken';
+use Carp 'croak';
+
+has 'connection';
+
+sub DESTROY {
+  my $self = shift;
+  return unless my $c = $self->connection;
+  return unless my $mysql = $self->mysql;
+  $mysql->_enqueue($c);
+}
+
+sub backlog { scalar @{shift->{waiting} || []} }
+
+sub begin {
+  my $self = shift;
+  $self->query('START TRANSACTION');
+  $self->query('SET autocommit=0');
+  my $tx = Mojo::mysql::Native::Transaction->new(db => $self);
+  weaken $tx->{db};
+  return $tx;
+}
+
+sub connect {
```

Comment:
I have a different plan for "url". To support failover hosts just like the JDBC connector for mysql.
http://dev.mysql.com/doc/connector-j/en/connector-j-reference-configuration-properties.html
If this functionality is added then "url" cannot be passed to Connection or Database. (*again this is future plan*).